### PR TITLE
Update major tag

### PR DESCRIPTION
**Pull request by sbe-arg**
If you use dependabot in your actions. Checkov is the most annoying of all bumping patch versions almost everyday. Lets have a major tag update as well.

Bump major to facilitate 

```
      - name: checkov
        uses: bridgecrewio/checkov-action@v12
```
as well of
```
      - name: checkov
        uses: bridgecrewio/checkov-action@v12.1708.0
```

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -7,19 +7,52 @@ jobs:
...
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           WITH_V: true
-          ACTIONS_ALLOW_UNSECURE_COMMANDS: 'true'
+
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: '0'
+
+      - name: version-tag-mayor
```
Comment (ankon) [inline .github/workflows/build.yml:33]: Typo: "ma_j_or"

Comment (sbe-arg) [reply .github/workflows/build.yml]: Thanks for spoting that :vulcan_salute:

**Review by ankon — APPROVED** [review]
LGTM :)

**Comment by sbe-arg** [comment]
@nimrodkor can I get some feedback? 

We can alternativelly use the native tag action but it does not automacilly find out the latest major

```
    - name: Bump version and push tag
      uses: anothrNick/github-tag-action@1.39.0
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        CUSTOM_TAG: v12
```
^ don't like that

https://github.com/anothrNick/github-tag-action/pull/157

**Comment by sbe-arg** [comment]
@schosterbarak anyone around?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -7,19 +7,50 @@ jobs:
...
+            echo "No new commits since previous tag. Skipping..."
+            exit 0
+          fi
+
+          [[ "$tag" =~ ^(v[0-9]+) ]]
+          major=${BASH_REMATCH[1]}
+
+          # update major tag
+          git tag -f "$major"
+          git push -f --tags
```
Comment (gruebel) [inline .github/workflows/build.yml:56]:
```suggestion
          git push -f origin "$major"
```
no need to force push everything

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -7,19 +7,50 @@ jobs:
...
+          # get current commit hash for tag
+          tag_commit=$(git rev-list -n 1 ${tag})
+          # get current commit hash
+          commit=$(git rev-parse HEAD)
+
+          if [ "$tag_commit" == "$commit" ]
+          then
+            echo "No new commits since previous tag. Skipping..."
+            exit 0
+          fi
```
Comment (gruebel) [inline .github/workflows/build.yml:49]: you could remove this and make the triggering of this step simpler by leveraging the output of the step `stefanzweifel/git-auto-commit-action`
https://github.com/stefanzweifel/git-auto-commit-action#outputs

**Review by gruebel — COMMENTED** [review]
thanks for the PR, I added some suggestions

File: .github/workflows/build.yml (hunk 2 above)
Comment (sbe-arg) [reply .github/workflows/build.yml]: Good idea!

File: .github/workflows/build.yml (hunk 3 above)
Comment (sbe-arg) [reply .github/workflows/build.yml]: Actually good point I did not see the outputs of the previous job.
Have addressed accordigly.

**Comment by sbe-arg** [comment]
Thanks @gruebel for the feedback

**Review by gruebel — APPROVED** [review]
nice, great job 🥇

**Pull request merged**